Modify check for ADB public key

dilruacs opened the pull request:
## Description:

Remove check for the public key if `adbkey` is being used. The public key is not needed to connect to the device, only the private key.
`adb` from platform-tools 28.0.2 does not even generate a `adbkey.pub` file anymore.

**Related issue (if applicable):** fixes #22377

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#9054

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

dilruacs committed:
Remove check for public key

JeffLIrion commented:
I think you could remove the `has_adb_files` function altogether and simply use `cv.isfile` in the schema, but I'm not positive about that. Also, could you please update the documentation, as there are some references to the adbkey.pub file. 

And thanks for submitting a pull request to fix the issue!

dilruacs committed:
Remove has_adb_files, directly call cv.isfile

JeffLIrion commented:
Have you tested this code? I don't think it will work. 

https://github.com/JeffLIrion/python-androidtv/blob/345507ffd7a07c97d8ee6c318abaa9aed8eca7e2/androidtv/basetv.py#L25

https://github.com/google/python-adb/blob/d9b94b2dda555c14674c19806debb8449c0e9652/adb/sign_pythonrsa.py#L63

dilruacs commented:
I first tested with an empty file (`touch adbkey.pub`), as a public key isn't generated anymore by ADB.
After removing the check for the public key, everything was working correctly as the dummy file still existed.

You're right, the upstream library still checks for the existence of the public key although an empty file does not cause problems.
Removing the file completely causes a `error: No such file or directory`.

As the real issue is still a problem (missing `adbkey.pub` when using the latest platform-tools) and a public key isn't really needed for the connection, I request to leave this PR open until a workable solution is found upstream.
I will then modify this PR.

dilruacs commented:
`python-adb` is using the Python `rsa` library for it's RSA functions.
The `Crypto.PublicKey` library is able to determine the public key from the private key using the function `publickey()` on the object.
I will have to ponder on the "Right Way" to solve this dilemma. The `adb` binary provided with `platform-tools` apparently deduces the public key from the generates private key also, as it doesn't store it separately.

JeffLIrion commented:
> I first tested with an empty file (`touch adbkey.pub`), as a public key isn't generated anymore by ADB.
> After removing the check for the public key, everything was working correctly as the dummy file still existed.
> 
> You're right, the upstream library still checks for the existence of the public key although an empty file does not cause problems.
> Removing the file completely causes a `error: No such file or directory`.
> 
> As the real issue is still a problem (missing `adbkey.pub` when using the latest platform-tools) and a public key isn't really needed for the connection, I request to leave this PR open until a workable solution is found upstream.
> I will then modify this PR.

Thanks for checking that! 

Given that `adbkey.pub` is needed, even if it's just an empty file, I think that one of the following should be included with this pull request:

* Create an empty `adbkey.pub` file if one does not exist. 
* Add documentation and/or a logging message instructing the user to create an `adbkey.pub` file if one does not exist. 
* Modify the `adb` package so that `adbkey.pub` is not required. 

Unfortunately, the `adb` package has pretty much been left for dead. It also doesn't work well, if at all, for newer devices. I had to fork the package and put a release on pypi because the version on there hadn't been updated in so long. I'm not looking to maintain the package, so the Python ADB implementation option is more of a legacy option, and if it doesn't work for someone's device then the advice is to use the ADB server option.

dilruacs committed:
Check for missing adbkey.pub, create dummy if not found

dilruacs commented:
- Updated the documentation of -current to offer the workaround of creating an empty public key (`touch adbkey.pub`)
- Modified `has_adb_files()` so it creates a dummy public key (empty file) if it doesn't find one

I am unsure if the solution I used is the most elegant one (`import os` in the component?)

dilruacs committed:
Reorder imports

JeffLIrion commented:
I haven't tested this and I'm not sure if the exception `FileNotFound` is correct (or if it's different between Python 2 and 3), but I think this patch to `androidtv` will allow for the public key to not exist. 

https://github.com/JeffLIrion/python-androidtv/pull/31

dilruacs committed:
Merge remote-tracking branch 'upstream/dev' into dev

dilruacs committed:
Merge remote-tracking branch 'upstream/dev' into dev

JeffLIrion commented:
I just pushed `androidtv` v0.0.14, which doesn't require the public key. Now all you need to do is bump the version and remove the check for the public key file.

dilruacs committed:
Bumped androidtv library version, deactivated pubkey test

dilruacs committed:
Code works without pubkey, removed function

dilruacs committed:
Merge remote-tracking branch 'upstream/dev' into dev

dilruacs commented:
@JeffLIrion Can confirm. Bumped `androidtv` to version 0.0.14 and removed the `has_adb_files()` function.

Tested and works! Thanks for looking into this problem with me.

dilruacs committed:
Removed "import os", not needed anymore

JeffLIrion commented:
No problem!

You also need to bump the version in requirements_all.txt (in the root folder of the repo).

dilruacs committed:
Bump library version
